hedera-node/src/main/java/com/hedera/services/contracts/execution/CallLocalEvmTxProcessor.java

```diff
@@ -98,13 +108,18 @@ protected MessageFrame buildInitialFrame(
 			final Address to,
 			final Bytes payload
 	) {
-		Bytes code = updater.get(to).getCode();
-		return baseInitialFrame
-				.type(MessageFrame.Type.MESSAGE_CALL)
-				.address(to)
-				.contract(to)
-				.inputData(payload)
-				.code(new Code(code, Hash.hash(code)))
-				.build();
+		try {
+			Code code = codeCache.get(to);
+			return baseInitialFrame
+					.type(MessageFrame.Type.MESSAGE_CALL)
+					.address(to)
+					.contract(to)
+					.inputData(payload)
+					.code(code)
+					.build();
+		} catch (RuntimeException e) {
+			logger.warn("Error fetching code from cache", e.getCause());
```

Comment:
As above, this situation is likely serious enough to merit capturing the full stack trace.